tests/System.Slices.Tests/ReferenceCounterTests.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System.Runtime;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Slices.Tests
+{
+    public class ReferenceCounterTests
```

Comment:
Add a test that hits the resize code paths, both for ReferenceCounter and for ObjectTable. 

The tests for both pieces should also be separate.